RecFCCeeCalorimeter/src/components/CalibrateCaloClusters.cpp

```diff
@@ -0,0 +1,368 @@
+#include "CalibrateCaloClusters.h"
+
+// Key4HEP
+#include "k4Interface/IGeoSvc.h"
+
+// FCC Detectors
+#include "detectorCommon/DetUtils_k4geo.h"
+
+// DD4hep
+#include "DD4hep/Detector.h"
+
+// our EDM
+#include "edm4hep/Cluster.h"
+#include "edm4hep/ClusterCollection.h"
+#include "edm4hep/CalorimeterHitCollection.h"
+
+DECLARE_COMPONENT(CalibrateCaloClusters)
+
+// convert vector data with given shape into ONNX runtime tensor
+template <typename T>
+Ort::Value vec_to_tensor(std::vector<T> &data, const std::vector<std::int64_t> &shape)
+{
+  Ort::MemoryInfo mem_info =
+      Ort::MemoryInfo::CreateCpu(OrtAllocatorType::OrtArenaAllocator, OrtMemType::OrtMemTypeDefault);
+  auto tensor = Ort::Value::CreateTensor<T>(mem_info, data.data(), data.size(), shape.data(), shape.size());
+  return tensor;
+}
+
+CalibrateCaloClusters::CalibrateCaloClusters(const std::string &name,
+                                             ISvcLocator *svcLoc)
+    : GaudiAlgorithm(name, svcLoc),
+      m_geoSvc("GeoSvc", "CalibrateCaloClusters")
+{
+  declareProperty("inClusters", m_inClusters,
+                  "Input cluster collection");
+  declareProperty("outClusters", m_outClusters,
+                  "Calibrated (output) cluster collection");
+}
+
+StatusCode CalibrateCaloClusters::initialize()
+{
+  // Initialize base class
+  {
+    StatusCode sc = GaudiAlgorithm::initialize();
+    if (sc.isFailure())
+    {
+      return sc;
+    }
+  }
+
+  // Check if readouts exist
+  {
+    bool readoutMissing = false;
+    for (size_t i = 0; i < m_readoutNames.size(); ++i)
+    {
+      auto readouts = m_geoSvc->getDetector()->readouts();
+      if (readouts.find(m_readoutNames.value().at(i)) == readouts.end())
+      {
+        readoutMissing = true;
```

Comment:
You can directly return `StatusCode::FAILURE` and not use `readoutMissing` variable.

Reply:
fixed in latest commit